packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx

```diff
@@ -223,6 +238,10 @@ export default class TooltipPopper extends React.Component<Props> {
                 right: popperProps.style.right,
                 position: popperProps.style.position,
                 transform: popperProps.style.transform,
+                // We hide the content if the popper isn't ready yet. This
+                // makes it so users do not see the tooltip in the wrong place
+                // while it re-positions itself
+                visibility: !isReady ? "hidden" : undefined,
```

Comment:
**question:** I wonder if instead of relying on React's state, we could use a Class variable to avoid re-rendering this component for this case? not sure if it's possible and/or will make an impact, but wanted to check with you about this approach.

Reply:
Good question, I didn't think of this approach! I tried it and the TooltipPopper doesn't render the contents as expected. This is because the `_renderPositionedContent` function is never called once `this.isReady` is `true`. So the additional re-render from the update to React's state is needed so that it triggers the function to be called again to remove the visibility styling

Reply:
Ah I see, thanks for the explanation.